### DISCUSSION TITLE
Is there a way to refresh the token when using react provider context?

### DISCUSSION THREAD

**Discussion by ekiim** [fief-dev/fief#418] [Q&A]
I'm struggling to find the proper way to handle the token refreshing when using the provider?

I am wrapping most of the behavior of the fief client in my own hook (so It integrates with my API that require a fief token), but when doing testing, I find several times, in a situation that the token expired, and have to login-logbackon, very often which feels like bad UI, so I'm trying to find the condition to use your `useFiefTokenInfo` and invoke manually new request for token?

If the library does not include it, I might need to do it manually.

Is this something I'm missing? am I doing something wrong? or it's just not implemented?

**Comment by fief-bailiff** [comment]
Hail, @ekiim 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to [star our repository](https://github.com/fief-dev/fief) ⭐️

<p align="center">
<img width="656" alt="star-fief" src="https://github.com/fief-dev/fief/assets/1144727/b2db7f59-102f-47e5-9751-67fa10372c52">
</p>

## Want to support us?

Subscribe to one of our paid plan to help us continue our work and receive exclusive information and benefits! Starts at $5/month 🪙

<p align="center">
    <a href="https://polar.sh/fief-dev">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/embed/subscribe.svg?org=fief-dev&darkmode=1">
      <img alt="Subscribe" src="https://polar.sh/embed/subscribe.svg?org=fief-dev">
    </picture>
    </a>
</p>

Farewell!

**Comment by frankie567** [comment]
Fief supports the concept of `refresh_token` besides the `access_token` when authenticating, if you ask for the `offline_scope`. But it's true that the JS/React SDK lacks proper support for it. I'll add it to the backlog!

Meanwhile, you can increase the lifetime of access tokens to limit the problem: https://docs.fief.dev/configure/clients/#authorization-code-and-tokens-lifetime

**Reply by jirhiker** [reply]
Im using version 0.29 and to get a `refresh_token`  I used `offline_access` as a scope. 

Then with the `refresh_token` in hand I was able to get a new access_token using
```
const refresh_token = authstate?.tokenInfo.refresh_token
const [tokenInfo, userinfo] = await fiefClient.authRefreshToken(refresh_token)
sessionStorage.setItem('fief-authstate', JSON.stringify({tokenInfo, userinfo}))
```
